datovka 4.15.6 -> 4.21.1

ovlach opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

update: datovka - https://gitlab.nic.cz/datovka/datovka/-/blob/v4.21.1/ChangeLog 
new package: https://gitlab.nic.cz/datovka/libdatovka (datovka now depends on libdatovka)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

ovlach committed:
datovka: 4.15.6 -> 4.21.1

https://gitlab.nic.cz/datovka/datovka/-/blob/v4.21.1/ChangeLog

AndersonTorres commented on maintainers/maintainer-list.nix:
```diff
@@ -10770,6 +10770,12 @@
     githubId = 15930073;
     name = "Moritz Scheuren";
   };
+  ovlach = {
```
split this file in a separate commit.

ovlach replied:
commit was splitted

AndersonTorres commented on pkgs/applications/networking/datovka/default.nix:
```diff
@@ -2,30 +2,31 @@
     description = "Client application for operating Czech government-provided Databox infomation system";
     homepage = "https://www.datovka.cz/";
-    license = licenses.lgpl3;
+    license = licenses.gpl3;
```
```suggestion
    license = licenses.gpl3Plus;
```

ovlach replied:
changed

AndersonTorres commented on pkgs/development/libraries/libdatovka/default.nix:
```diff
@@ -0,0 +1,42 @@
+  meta = with lib; {
+    description = "Client library for accessing SOAP services of Czech government-provided Databox infomation system";
+    homepage = "https://gitlab.nic.cz/datovka/libdatovka";
+    license = licenses.gpl3;
```
```suggestion
    license = licenses.gpl3Plus;
```

ovlach replied:
changed

ovlach committed:
maintainers: add ovlach

ovlach committed:
libdatovka: init at 0.2.1

datovka application now depends on libdatovka(https://gitlab.nic.cz/datovka/datovka/-/blob/v4.16.0/ChangeLog).
